### PR TITLE
ledger-on-sql: Use random log entry ID allocation so we don't depend on `SeedService`. [KVL-1002]

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I am trying to move `SeedService` to the _participant-integration-api_ Bazel package, and this is getting in my way.

To make this work, I created a trait, `LogEntryIdAllocator`, and implemented it using a `SeedService` in _sandbox-next_. The _ledger-on-sql/app_ variant uses `UUID.randomUUID()` instead, as this is only used in testing.

I also cleaned up a little around `SubmissionValidatorSpec`, as I was in the neighborhood.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+trait LogEntryIdAllocator {
+  def allocate(): DamlLogEntryId
+}
+
+object LogEntryIdAllocator {
+  object Random extends LogEntryIdAllocator {
+    override def allocate(): DamlLogEntryId =
+      DamlLogEntryId.newBuilder
+        .setEntryId(ByteString.copyFromUtf8(UUID.randomUUID().toString))
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala:19]: You may want to use the thread local variant to reduce contention.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala [hunk 2]
```diff
@@ -318,49 +315,26 @@ object SubmissionValidator {
       InputState,
   ) => LoggingContext => LogEntryAndState
 
-  def create[LogResult](
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala:321]: I take it's not used anywhere (apart from tests)

File: ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala [hunk 3]
```diff
@@ -26,13 +26,12 @@ abstract class SqlLedgerReaderWriterIntegrationSpecBase(implementationName: Stri
...
+      ledgerId = ledgerId,
+      participantId = participantId,
+      metrics = metrics,
       engine = Engine.DevEngine(),
-      jdbcUrl(testId),
+      jdbcUrl = jdbcUrl(testId),
       resetOnStartup = false,
-      // Using a weak random source to avoid slowdown during tests.
-      seedService = SeedService(Seeding.Weak),
+      logEntryIdAllocator = LogEntryIdAllocator.Random,
```
Comment (fabiotudone-da) [inline ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala:35]: Not really related to this PR, but I am worried that it's not apparent anywhere in the code that this impl. is used only for testing (is it?).

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala (hunk 2 above)
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala:321]: Exactly, yes.

File: ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala:35]: Kind of; depends how you define "testing". This code is only used by _ledger-on-sql_'s own `Main` entrypoint (not Sandbox), and so therefore it's only used in tests.

As this implementation is only used in _ledger-on-sql_, I'll just move it there and rename it accordingly.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala (hunk 1 above)
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala:19]: As it's only used in tests, I'm going to not bother. :upside_down_face:

File: ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala]: Moved to _ledger-on-sql/app_.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala (hunk 1 above)
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdAllocator.scala]: Alright, I propose to add a comment to let readers know.

File: ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala (hunk 3 above)
Comment (fabiotudone-da) [reply ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala]: OK, now I am only worried that it's not apparent anywhere in the code that `ledger-on-sql` is only used for testing (is it?) :)

Comment (SamirTalwar-DA) [reply ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala]: _ledger-on-sql/app_ is only used for testing; that's where I moved it.

**Review by hubert-da — APPROVED** [review]

File: ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/ledger-on-sql/src/test/lib/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriterIntegrationSpecBase.scala]: Pushed a _README.md_.

**Review by miklos-da — APPROVED** [review]

**Pull request merged**